src/rust/receiver-mock/src/router.rs

```diff
@@ -438,10 +440,23 @@ pub async fn handler_receiver(
         // Logs & events
         "application/x-www-form-urlencoded" => {
             {
-                let mut log_repository = app_state.logs.write().unwrap();
+                let mut message_count = 0;
+                let mut byte_count = 0;
+                let mut log_messages = app_state.log_messages.write().unwrap();
                 for line in lines.clone() {
-                    log_repository.add_log_message(line.to_string(), remote_address)
+                    message_count += 1;
+                    byte_count += line.len() as u64;
+                    if opts.store_logs {
+                        match log_messages.add_log_message(line.to_string()) {
+                            Ok(_) => (),
+                            Err(error) => {
+                                eprintln!("{} when processing log line {}", error, line);
+                            }
+                        }
+                    }
                 }
+                let mut log_stats = app_state.log_stats.write().unwrap();
+                log_stats.update(message_count, byte_count, remote_address);
```

Comment:
Can we extract this block similarly as it's done for metrics, e.g. `metrics::handle_prometheus( ... )` ?

Reply:
Eh, it's a bit tough because we're barely doing any parsing here, and the storage parts are owned by appstate. I'll move it to an appstate method instead.